tests/transfer-item.test.js

```diff
@@ -0,0 +1,112 @@
+
+            beforeAll(async () => {
+                jest.setTimeout(JEST_TIMEOUT);
+                oldInventoryResponse = await natsConnection.request(
+                    "tokenization-service.create_inventory",
+                    undefined,
+                    { timeout: JEST_TIMEOUT }
+                );
+
+                newInventoryResponse = await natsConnection.request(
+                    "tokenization-service.create_inventory",
+                    undefined,
+                    { timeout: JEST_TIMEOUT }
+                );
+
+                oldInventoryAddress = jsonCodec.decode(
+                    oldInventoryResponse.data
+                ).inventory_address;
+
+                newInventoryAddress = jsonCodec.decode(
+                    newInventoryResponse.data
+                ).inventory_address;
+            });
+
+            describe("When I assign this item to the old inventory", () => {
+                beforeAll(async () => {
+                    jest.setTimeout(JEST_TIMEOUT);
+                    response = await natsConnection.request(
+                        "tokenization-service.assign_inventory_item",
+                        jsonCodec.encode({
+                            inventory_address: oldInventoryAddress,
+                            item_id: newItemId,
+                            instance_number: 1
+                        }),
+                        { timeout: JEST_TIMEOUT }
+                    );
+                });
+
+                it("Then transfers the item to the new Inventory", () => {
+                    beforeAll(async () => {
+                        response = await natsConnection.request(
+                            "tokenization-service.transfer_inventory_item",
+                            jsonCodec.encode({
+                                old_inventory_address: oldInventoryAddress,
+                                new_inventory_address: newInventoryAddress,
+                                instance_number: 1,
+                                item_id: newItemId
+                            })
+                        );
+
+                        describe("When I retrieve the item from the new inventory", () => {
+                            beforeAll(async () => {
+                                response = await natsConnection.request(
+                                    "tokenization-service.get_inventory_item",
+                                    jsonCodec.encode({
+                                        inventory_address: newInventoryAddress,
+                                        item_id: newItemId,
+                                        instance_number: 1
+                                    })
+                                );
+                            });
+
+                            it("Then returns the item data", () => {
+                                expect(jsonCodec.decode(response.data)).toEqual(
+                                    {}
```

Comment:
actually, one last thing, but we should set some data in the inventory item in the blockchain before transferring to the new inventory, and verify here that the data is properly set after the transfer.